File: backend_api/CalendarShop.Api/Models/Review.cs
namespace CalendarShop.Api.Models;

public class Review
{
    public int ReviewId { get; set; }
    public int UserId { get; set; }
    public int ProductId { get; set; }
    public int OrderId { get; set; }
    public int OrderItemId { get; set; }
    public int Rating { get; set; }
    public string? Comment { get; set; }
    public string Status { get; set; } = "Visible";
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public DateTime? UpdatedAt { get; set; }

    public User? User { get; set; }
    public Product? Product { get; set; }
    public Order? Order { get; set; }
    public OrderItem? OrderItem { get; set; }
}

File: mobile_flutter/lib/features/review/domain/entities/review.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'review.freezed.dart';

@freezed
class Review with _$Review {
  const factory Review({
    required int reviewId,
    required int userId,
    required String userFullName,
    String? userAvatarUrl,
    required int productId,
    required String productName,
    required int orderItemId,
    required int rating,
    String? comment,
    required String status,
    required DateTime createdAt,
    DateTime? updatedAt,
  }) = _Review;
}
